src/ts/index.ts

```diff
@@ -1,5 +1,46 @@
 import { ethers, BigNumberish, Signature, SignatureLike, Wallet } from "ethers";
 
+/**
+ * EIP-712 domain data used by GPv2.
+ *
+ * Note, that EIP-712 allows for an extra `salt` to be added to the domain that
+ * isn't used.
+ * <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator>
+ */
+export interface EIP712Domain {
+  /**
+   * The user readable name of signing domain.
+   */
+  name: string;
+  /**
+   * The current major version of the signing domain.
+   */
+  version: string;
+  /**
+   * The EIP-155 chain ID.
+   */
+  chainId: number;
+  /**
+   * The address of the contract that will verify the EIP-712 signature.
+   */
+  verifyingContract: string;
+}
+
+/**
+ * Return the Gnosis Protocol v2 domain used for EIP-712 signing.
+ */
+export function domain(
+  chainId: number,
+  verifyingContract: string,
```

Comment:
ubernit: Maybe include the fact that this is an address (and not the bytecode or contract name) by appending Address to the variable name.

Reply:
I prefer to keep the name the same as it appears in the standard, as I expect that would be more aligned with people's expectations. What I think we should do is add documentation that specifies this is an address (as well as verify that this is indeed an address).